### ISSUE TITLE
Track Action Page views vs Pitch Page views

### ISSUE THREAD

**Issue by aaronschachter** [DoSomethingArchive/legacy-website#3308]
**ISSUE:** We need to track unique "Action Page" views in Google Analytics.  Current limitation is that we don't have a way to distinguish page views between the Pitch Page and Action Page, since both have the same URL.

**UPDATED:** Fire a GA tracking event for every action page view, in order to specifically track action page views.  We could fire one for Pitch Page views too, for the hay of it.

**ORIGINAL thought:**

e.g. Use a query string with redirects: `/campaigns/teens-jeans?action=signup`
- When a user signs up, direct them to the campaign node with `?action=signup` appended to query string
- If a user is viewing a link with `?action=signup` and is actually not signed up (like if someone shares their `?signedup` link), redirect to URL without the query string

**Comment by aaronschachter** [comment]
One thing that kinda sucks about this is that if a user shares the action page, this screws the analytics up.

**Comment by weerd** [comment]
Anything we could do with HTTP headers? Not even sure if that would solve anything, but just throwing it out there.

**Comment by DFurnes** [comment]
Can we check for `?signedup` query string when rendering the pitch page and redirect user to a URL without it if they aren't authenticated & signed up?

Something like this?

``` php
if($_GET['signedup']) { 
   header("Location: http://www.dosomething.org/campaigns/blahblah");
   die();
}
```

**Comment by angaither** [comment]
We could have the pitch page url we use internally be exposed (or have a pretty version of it exposed)
That way that can be shared, or the main url can be shared and application logic already handles redirecting depending on the user's status.

**Comment by DFurnes** [comment]
Also there's probably a more appropriate Drupal-y way to do that, so ignore my PHP. :sheep:

**Comment by aaronschachter** [comment]
Yeah I'm into the redirect (its `drupal_goto` btw @DFurnes )

@angaither The separate pitch page URL is a good idea too... but it feels like a bigger lift here if the redirect gets us what we'd need.  We'd to need to figure out the pretty version of the pitch page URL, and staff / users might get a little tripped up by having 2 separate URLs...

**Comment by mshmsh5000** [comment]
@aaronschachter @DFurnes @angaither The redirect will only work if the app gets hit...just thinking through this.

`?signedup` pages will always be generated by authenticated users who sign up, which means they won't get cached in Varnish (because of the session cookie).

So if an authenticated user shares the `?signedup` link with an unauthenticated user, the new user will hit `?signedup`, which will hit the app (because it's not in Varnish), which will respond with a 302 redirect. Varnish will cache the redirect.

At that point, Varnish should always have the 302 redirect cached, and authenticated users should never see the redirect, because Varnish sees their session cookies and passes them through.

So basically :+1: and no value added from this guy

**Comment by mshmsh5000** [comment]
@joshcusano @Bladt This is looking a little more complicated than it needs to be.

What if the action view always fires off a GA event like "Teens For Jeans - Signup" gets fired? Then you can just track how many of those events occur. They'll never fire on the pitch page because the JS code for this will only be in the action view template.

The event label can be determined in the action template, and basically be `[CAMPAIGN TITLE] - Signup`.

**Comment by aaronschachter** [comment]
cc @DFurnes -- I know you guys were chatting GA events.

**Comment by DFurnes** [comment]
The issue that we were seeing is that events that were hooked up form submission were firing inconsistently. My best guess is that browsers (and from the data, it seems like in particular mobile browsers) were canceling the analytics event HTTP request when the page changes since we submit this request right before a new page is loaded.

**Comment by aaronschachter** [comment]
So the idea here is we just simply send it when specifically the Action Page template -- not trying to tie any of the events to form submission.

**Comment by DFurnes** [comment]
Oh, I totally misread that. Oops. :sweat_smile:

We can do that if that's simpler. So the plan is to conditionally include the code to fire the GA event when a user sees the action page for the first time?

**Comment by aaronschachter** [comment]
I think it can just be every time the action page is loaded, fire the GA event for Action page, if we just want to track Action Page views. (vs first page views)

Could do the same for a different Pitch page event, if we wanted to, to keep it similar.

**Comment by DFurnes** [comment]
That's fine too. Didn't we want to be able to specifically track signups (first action page view, as opposed to all action page views) though?

**Comment by aaronschachter** [comment]
@Bladt can you chime in here? sounded like tracking just the Action Page would give us 90% of what we need, which sounds pretty good for Tuesday's launch.

**Comment by DFurnes** [comment]
From discussion at standup: we'll be triggering `[CAMPAIGN TITLE] - Action Page View` event on every page view on the Action Page for a given campaign.

**Comment by Bladt** [comment]
@DFurnes what will the resulting url structure be? (working on setting up automatic reporting)

**Comment by DFurnes** [comment]
The URL will be the same, but we'll be firing a Custom Event on each page load of the action page (rather than triggering it on pitch page form submission as we were before).

**Comment by Bladt** [comment]
Ok. ETA what this change will make it to production?

**Comment by DFurnes** [comment]
Will be up with next deploy, which is slated to go out tomorrow morning.
